djangosaml2/views.py

```diff
@@ -177,17 +175,20 @@ def login(request,
         return HttpResponseRedirect(get_location(result))
     elif binding == BINDING_HTTP_POST:
         if not post_binding_form_template:
+            # use the html provided by pysaml2
             return HttpResponse(result['data'])
-        try:
-            params = get_hidden_form_inputs(result['data'][3])
+        else:
+            # manually get request XML to build our own template
+            request_id, request_xml = client.create_authn_request(
```

Comment:
Oh no, it's still very broken if you have any characters in your IDP SSO url that choke an xml parser.   I only fixed the & because the was one we had happen in the field, and I intended it only as a temporary fix until I dug more into pysaml2.   

I can change it so it does only one or the other instead of both in that case if you like, but explicitly getting the parameters instead of parsing them from html that pysaml2 generates is going to be far less bug-prone.  Do you want me to go ahead and have it only do one or the other?